cartoframes/viz/kuviz.py

```diff
@@ -19,28 +15,33 @@
 class KuvizPublisher:
     def __init__(self, credentials=None):
         self.kuviz = None
-        self._maps_api_key = DEFAULT_PUBLIC
-
-        self._credentials = credentials or get_default_credentials()
-        self._auth_client = _create_auth_client(self._credentials)
         self._layers = []
+        self._auth_client = _create_auth_client(credentials)
+        self._auth_api_client = _create_auth_api_client(credentials)
 
     @staticmethod
     def all(credentials=None):
-        auth_client = _create_auth_client(credentials or get_default_credentials())
+        auth_client = _create_auth_client(credentials)
         kmanager = _get_kuviz_manager(auth_client)
         kuvizs = kmanager.all()
         return [kuviz_to_dict(kuviz) for kuviz in kuvizs]
 
     def get_layers(self):
         return self._layers
 
-    def set_layers(self, layers, name, table_name=None):
-        table_name = table_name or '{}_{}_table'.format(name, int(time.time() * 1000))
+    def set_layers(self, layers, maps_api_key=None):
+        if not isinstance(maps_api_key, str):
+            maps_api_key = self._create_maps_api_keys(layers)
+
+        self._layers = []
+        for layer in layers:
+            layer_copy = copy.deepcopy(layer)
+
+            # Set the Maps API key if the source is private
+            if not layer_copy.source.is_public() and layer_copy.credentials is not None:
+                layer_copy.credentials['api_key'] = maps_api_key
```

Comment:
does it work with base_url only? I don't remember well now, but we were doing more things in `_add_layers_credentials`

Reply:
Yes. `_add_layers_credentials` was doing the same as Source.get_credentials does. So the behavior is the same